### ISSUE TITLE
Support for sqlite backend

### ISSUE THREAD

**Issue by phosphene47** [mikecao/umami#245]
Hi! Thanks for making this great analytics app.

It would be great to see an SQLite database backend in addition to MySQL and PostgreSQL. SQLite is suitable for light workloads or when installing a database server isn't necessarily feasible.

From what I've observed, umami uses Prisma 2 for database connection which natively supports SQLite already. It would be a matter of simply adding a Prisma schema for it, modifying a few raw queries and distributing another Docker image.

**Comment by mikecao** [comment]
Yes, it would be possible, just not a high priority right now. I would welcome a PR from anyone who wants to tackle it.

**Comment by leirock** [comment]
I notice that @philippdormann's [modded version of umami](https://github.com/philippdormann/umami-sqlite) supports a local SQLite database. But seems no PR was created?

**Comment by philippdormann** [comment]
hi @leirock 
yes, it does support SQLite and worked last time I checked.

I stopped working on this due to time constraints and the (for my use case too high to justify) memory usage of umami.
afaik there were some major changes in v1.17.0 in terms of Prisma/ Next.js version bumps. Those dependency bumps might require some changes for sqlite to work 🤷‍♂️

see https://github.com/mikecao/umami/pull/680#issuecomment-909228819 for the PR 😉

note: I might get around to a proper PR for sqlite support with the current version. recently got some more time :)

**Comment by leirock** [comment]
hi @philippdormann 
Thanks very much for your reply. It's good news to hear that you got some more time recently. Cheers! 😂
